internal/app/wwctl/overlay/edit/main.go

```diff
@@ -115,10 +115,17 @@ func CobraRunE(cmd *cobra.Command, args []string) error {
 		}
 	}
 
+	// try renaming the tempfile to overlayfile first
 	err := os.Rename(tempFile.Name(), overlayFile)
 	if err != nil {
-		wwlog.Error("Unable to update %s: %s", overlayFile, err)
-		os.Exit(1)
+		// if it fails, which probably means that they exists on different partitions
+		// fallback to data copy
+		wwlog.Debug("Unable to rename temp file: %s to overlay file: %s, try copying the data", tempFile.Name(), overlayFile)
+		cerr := util.CopyFile(tempFile.Name(), overlayFile)
+		if cerr != nil {
+			wwlog.Error("Unable to copy data from temp file: %s to target file: %s, err: %s", tempFile.Name(), overlayFile, err)
+			os.Exit(1)
```

Comment:
Why not return the error here? `os.Exit(1)` is always not nice.
Also we should remove the `tempFile` afterwards.

Reply:
thanks for the comment!
I saw that previous code is using `os.Exit`, anyway, I've changed to `return error` in this case.
tempfile is removed here
https://github.com/warewulf/warewulf/blob/b61e3918908bd48ee7bb30af659fe946cc06efac/internal/app/wwctl/overlay/edit/main.go#L69C2-L69C8